pkg/operators/v1/operatorgroup_types.go

```diff
@@ -45,6 +45,49 @@ type OperatorGroupSpec struct {
 	// Static tells OLM not to update the OperatorGroup's providedAPIs annotation
 	// +optional
 	StaticProvidedAPIs bool `json:"staticProvidedAPIs,omitempty"`
+
+	UpgradeStrategy *UpgradeStrategy `json:"upgradeStrategy,omitempty"`
```

Comment:
Since we have a default value for this setup in the CRD schema, WDYT about making this field always present, such that we can always apply the default `name: Default`?

```suggestion
	UpgradeStrategy UpgradeStrategy `json:"upgradeStrategy"`
```

I think the semantics of this plus the defaulting in the schema mean that an OG could be submitted without setting upgrade strategy but then when the user fetches it from the cluster, they'd see upgrade strategy populated with the default name.

Reply:
Thinking more: I wonder what this would mean for existing OperatorGroups?

Reply:
This change would invalidate OperatorGroups where the spec is defined like so:
```bash
$ k apply -f-<<EOF
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  generation: 3
  name: foo
  namespace: bar
spec: {}
EOF
error: error validating "STDIN": error validating data: ValidationError(OperatorGroup.spec): missing required field "upgradeStrategy" in com.coreos.operators.v1.OperatorGroup.spec; if you choose to ignore these errors, turn validation off with --validate=false
```

Marking this struct as optional allowed the object above to be created but did not generate the default value for the `og.spec.upgradeStrategy.Name` field:
```bash
$ apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"operators.coreos.com/v1","kind":"OperatorGroup","metadata":{"annotations":{},"generation":3,"name":"foo","namespace":"bar"},"spec":{}}
  creationTimestamp: "2022-04-05T18:50:07Z"
  generation: 1
  name: foo
  namespace: bar
  resourceVersion: "33408"
  uid: e0d101e9-fc6c-4664-9e09-5fa37372a693
spec: {}
status:
  lastUpdated: "2022-04-05T18:50:07Z"
  namespaces:
  - ""
 ```
We can still make the field a non-pointer, but we would probably want to provide a helper function that treats an unset `og.spec.upgradeStrategy.Name` as the `Default` upgrade strategy.

Reply:
This was inaccurate and has been updated in the latest version of the PR.